root/usr/sbin/hotsync-promote

```diff
@@ -33,13 +33,30 @@ sed -i 's/enabled = 0/enabled = 1/g' /etc/yum/pluginconf.d/nethserver_events.con
 
 /sbin/e-smith/restore-config
 
+if [[ $? == 0 ]]; then
+    echo "[NOTICE] Configuration restore completed!"
+else
+    echo "[ERROR] restore-config failed. Last error messages from /var/log/messages:  "
+    echo ""
+    grep -E '(ERROR|FAIL)' /var/log/messages | tail -15 | sed 's/^/    /'
+    echo ""
+    ((++exit_code))
+fi
+
 # Check if network roles need to be assigned
 flag=/var/run/.nethserver-fixnetwork
-echo "Configuration restore completed!"
 if [[ ! -f $flag ]]; then
     echo "To complete promotion to master, launch"
+    echo ""
+    echo "   /sbin/e-smith/signal-event post-restore-data"
+    echo ""
 else 
     echo "Go to Server Manager page \"Network\" and reassign roles to network interfaces"
     echo "and complete promotion to master by launching"
+    echo ""
+    echo "   /sbin/e-smith/signal-event post-restore-data"
+    echo ""
+    ((++exit_code))
```

Comment:
Why this? Is it an error?

Reply:
It's Bash arithmetic expansion syntax :) I'm counting the number of errors occurred, actually can be 0, 1, 2.

Reply:
But is it an error if network interfaces roles need to be reassigned?

Reply:
The command is interactive and exit code is probably not relevant.

However if it's used in another script I think it's more robust to report back a dirty state because a manual action is required. For instance, what happens if sysadmin writes:

    hotsync-promote && reboot